README.md

```diff
@@ -141,6 +141,41 @@ That includes warnings found by rustc (e.g. `dead_code`, etc.). If you want to a
 an error for Clippy warnings, use `#![deny(clippy::all)]` in your code or `-D clippy::all` on the command
 line. (You can swap `clippy::all` with the specific lint category you are targeting.)
 
+### Install from source
+
+If you are hacking on Clippy and want to install it from source, do the following:
+
+First, take note of the toolchain [override](https://rust-lang.github.io/rustup/overrides.html) in `/rust-toolchain`.
+We will use this override to install Clippy into the right toolchain.
+
+> Tip: You can view the active toolchain for the current directory with `rustup show active-toolchain`.
+
+From the Clippy project root, run the following command to build the Clippy executables (`clippy-driver` and
+`cargo-clippy`) and copy them into the toolchain directory. This may replace the currently installed clippy
+component.
+
+```terminal
+cargo build -Zunstable-options --out-dir "$(rustc --print=sysroot)/bin" --release
```

Comment:
This can be pretty destructive, perhaps we should recommend people make a copy toolchain and work within there?

Reply:
Also, won't this pollute that folder with rlib files too, or is it just the binary?

Reply:
> This can be pretty destructive, perhaps we should recommend people make a copy toolchain and work within there?

Clippy already uses a defined nightly toolchain. What's the chance that someone uses the exact same toolchain for something else? But a note explicitly stating, that this overrides the clippy of the toolchain SGTM.

> Also, won't this pollute that folder with rlib files too, or is it just the binary?

I just tested it. This only places the 2 binaries in the `--out-dir`

Reply:
> What's the chance that someone uses the exact same toolchain for something else?

Agreed. On the occasion that Clippy needs to be restored, its pretty easy with rustup, instructions below. But I can make this safer by adding `--bin cargo-clippy --bin clippy-driver`.

> But a note explicitly stating, that this overrides the clippy of the toolchain SGTM.

I already have that? But maybe it should be more explicit or have some bold.

Reply:
> This may replace the currently installed clippy component.

Let's just rewrite this to:

"This overrides the currently installed Clippy component"